### ISSUE TITLE
Install breaks on F29 system with `apt` installed

### ISSUE THREAD

**Issue by tildebyte** [junaruga/rpm-py-installer#182]
`apt` is not only **available** for Fedora 29, it is in the System "Everything" repo[1] (but not installed by default).
Using the presence of the `apt-get` command to determine whether or not a system is Debian-based seems fragile at best.

[1] e.g. on my system, [apt-0.5.15lorg3.95-33.git522.fc29.x86_64](http://mirror.math.princeton.edu/pub/fedora/linux/releases/29/Everything/x86_64/os/Packages/a/apt-0.5.15lorg3.95-33.git522.fc29.x86_64.rpm) (direct link to rpm package)

**Comment by junaruga** [comment]
@tildebyte thank you for your report!
I did not expect `apt-get` can be installed on the Fedora environment.
Right now `rpm-py-installer` judges the OS environment by checking if `apt-get` command exists on the `PATH` like below part.

https://github.com/junaruga/rpm-py-installer/blob/master/install.py#L1178-L1181

```
        if Cmd.which('apt-get'):
            linux = DebianLinux(python, rpm_path, **kwargs)
        else:
           linux = FedoraLinux(python, rpm_path, **kwargs)
```

Maybe we have to check by reading `/etc/fedora-release` or `/etc/fedora-release`.

**Comment by junaruga** [comment]
Possible temporary workflow is to rename `apt-get` command or remove it from the PATH..

**Comment by junaruga** [comment]
@tildebyte do you know good way to judge RPM base Linux (Actual package manager is RPM) and Debian base Linux (Actual package manager is apt)?

**Comment by tildebyte** [comment]
@junaruga I wish I did :S
Nothing comes to mind immediately, other than your suggestion of /etc/*-release...
Maybe something truly silly like seeing which command (apt or rpm) shows that the "kernel" package is installed :P

**Comment by junaruga** [comment]
@tildebyte Sorry for late response. I just realized we already have the logic with `/etc/os-release` in testing logic for the conditional tests considering Fedora base or Debian base OS.

So, I prefer to implement the way for actual logic as a solution for this ticket.
If you have a time, I am appreciate for your pull-request.
If you do not have a time, I will fix it after fixing multiarch related issue (https://github.com/junaruga/rpm-py-installer/issues/173, https://github.com/junaruga/rpm-py-installer/pull/192)

https://github.com/junaruga/rpm-py-installer/blob/master/tests/conftest.py#L24-L36

```
def _get_os_id():
    os_id = None
    if not os.path.isfile(OS_RELEASE_FILE):
        return os_id

    with open(OS_RELEASE_FILE) as f_in:
        for line in f_in:
            match = re.search(r'^ID=[\'"]?(\w+)?[\'"]?$', line)
            if match:
                os_id = match.group(1)
                break

    return os_id
```

**Comment by junaruga** [comment]
I like to check ID and ID_LIKE field considering below RPM base Linux distributions.

## Fedora

```
$ cat /etc/os-release
...
ID=fedora
...
```

## CentOS

```
$ podman run --rm -t centos:7 cat /etc/os-release
...
ID="centos"
ID_LIKE="rhel fedora"
...
```

## RHEL

```
$ podman run --rm -t registry.access.redhat.com/ubi7 cat /etc/os-release
...
ID="rhel"
ID_LIKE="fedora"
...
```

## OpenSUSE

```
$ podman run --rm -t opensuse cat /etc/os-release
...
ID=opensuse
ID_LIKE="suse"
...
```

## Amazon Linux

https://superuser.com/questions/291210/how-to-get-amazon-ec2-instance-operating-system-info

```
ID="amzn"
ID_LIKE="centos rhel fedora"
```

**Comment by junaruga** [comment]
I am working in progress for this issue now, sending a pull-request.

**Comment by junaruga** [comment]
Now I merged the PR to fix this issue.
I am going to release new version including it soon.
After that, if you see the issue, you can reopen.
Thanks.

**Comment by junaruga** [comment]
Now I released new version 0.9.1.
